File: Data Structures & Algorithms/reverse-string/submission-2.java
class Solution {
    public void reverseString(char[] s) {
        int l = 0, r = s.length - 1;
        char temp;
        temp = s[l];
        s[l++] = s[r];
        s[r--] = temp;
        while (l < r) {
            temp = s[l];
            s[l++] = s[r];
            s[r--] = temp;
        }
    }
}

File: Data Structures & Algorithms/house-robber/submission-2.rs
impl Solution {
    pub fn rob(nums: Vec<i32>) -> i32 {
        let (mut rob1, mut rob2) = (0, 0);
        for &num in &nums {
            let temp = (num + rob1).max(rob2);
            rob1 = rob2;
            rob2 = temp
        }
        rob2
    }
}
